yb-voyager/src/tgtdb/target_db_interface.go

```diff
@@ -38,18 +40,89 @@ type TargetDB interface {
 	IfRequiredQuoteColumnNames(tableName string, columns []string) ([]string, error)
 	ExecuteBatch(migrationUUID uuid.UUID, batch *EventBatch) error
 	GetDebeziumValueConverterSuite() map[string]tgtdbsuite.ConverterFn
-	GetEventChannelsMetaInfo(migrationUUID uuid.UUID) (map[int]EventChannelMetaInfo, error)
-	GetTotalNumOfEventsImportedByType(migrationUUID uuid.UUID) (int64, int64, int64, error)
-	InitLiveMigrationState(migrationUUID uuid.UUID, numChans int, startClean bool, tableNames []string) error
 	MaxBatchSizeInBytes() int64
 	RestoreSequences(sequencesLastValue map[string]int64) error
-	GetImportedEventsStatsForTable(tableName string, migrationUuid uuid.UUID) (*EventCounter, error)
-	GetImportedSnapshotRowCountForTable(tableName string) (int64, error)
 	GetGeneratedAlwaysAsIdentityColumnNamesForTable(table string) ([]string, error)
 	DisableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error
 	EnableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error
+
+	// NOTE: The following four methods should not be used for arbitrary query
+	// execution on TargetDB. The should be only used from higher level
+	// abstractions like ImportDataState.
+	Query(query string) (Rows, error)
+	QueryRow(query string) Row
+	Exec(query string) (int64, error)
+	WithTx(fn func(tx Tx) error) error
+}
+
+//=============================================================
+
+type Rows interface {
+	Row
+	Next() bool
+	Close()
+}
+
+type Row interface {
+	Scan(dest ...interface{}) error
+}
+
+type sqlRowsToTgtdbRowsAdapter struct {
+	rows *sql.Rows
```

Comment:
Could we just embed sql.Rows?

Reply:
Done. Redefined the `Close()` method for the reason mentioned in the previous answer.